src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/KestrelThread.cs

```diff
@@ -125,6 +133,34 @@ public void Stop(TimeSpan timeout)
             }
         }
 
+        private void DisposeConnections(TimeSpan timeout)
+        {
+            try
+            {
+                // Close and wait for all connections
+                if (!ConnectionManager.WalkConnectionsAndClose(timeout))
+                {
+                    _log.LogError(0, null, "Waiting for connections timed out");
```

Comment:
`_log.NotAllConnectionsClosedGracefully()` There's a risk where this could get logged multiple times where it wouldn't before, but I don't think that's too big of an issue.